Components es6 apply code review fix suggestions

Swiftb0y opened the pull request:
These are some of the minimally invasive fixes I suggested during code review. I don't have have access to my controller atm so it'll take some time until I'll be able to implement other stuff I suggested during review.

poelzi committed:
Use info not warning for skin COs

Be-ing committed:
CMake: remove obsolete references to files generated by scons

Be-ing committed:
rename maybeStartOrStopPolling to pollIfAnyControllersOpen

ywwg committed:
double click to reset knobs and sliders

Be-ing committed:
Components: rewrite with ES6 classes in version 0.1

Swiftb0y committed:
midi-components-0.1: "use-strict" for library

Swiftb0y committed:
midi-components-0.1: improve Button.types API (breaking)

Make Button.types enum a static frozen/read-only property on the
button object in an effort to reduce the exposure of the prototype

Swiftb0y committed:
midi-components-0.1: use arrow func where possible

replace `function(){}.bind(this)` pattern with arrow functions where
applicable.

Swiftb0y committed:
midi-components-0.1: simplify Component.trigger() & .disconnect()

Swiftb0y committed:
midi-components-0.1: reorder comparision with constants to be consistent

Be-ing committed:
Controller: make m_sDeviceName const

This will be required to use it in a QLoggingCategory which has a
deleted copy constructor.

Be-ing committed:
replace --controller-debug CLI argument with Qt logging categories

Logging is separated into several subcategores:
controller.CONTROLLERNAME: general messages
controller.CONTROLLERNAME.input: incoming data
controller.CONTROLLERNAME.output: outgoing data

CONTROLLERNAME is the lowercase device name with non-alphanumeric
characters removed.

This allows more fine tuned control of what messages are shown
in the console. This is required for the logging output to be
useful with controllers that continually send input such as the
Native Instruments Traktor Kontrol S4 Mk3 and Gemini GMX.

Be-ing committed:
Hercules DJ Console 4 Mx: use console.log instead of engine.log

This was the only script directly using engine.log instead of
the print wrapper in common-controller-scripts.js.

Be-ing committed:
HSS1394/PortMidiController: remove superfluous QString formatting

Be-ing committed:
MidiOutputHandler: correct comment

Be-ing commented on res/controllers/midi-components-0.1.js:
```diff
@@ -357,6 +358,7 @@
             options.group = "[Sampler" + options.number + "]";
             options.outKey = null, // hack to get Component constructor to call connect()
             super(options);
+            // this.connect();
```
Did you commit this on accident?

Swiftb0y replied:
Ups, yes this was on accident. It was left from a previous experiment where I tried to call connect after the parent constructor but that didn't work out-right so I wanted to take a look some other time.

Be-ing commented on res/controllers/midi-components-0.1.js:
```diff
@@ -125,6 +125,7 @@
             }
         }
         trigger() {
+
```
stray line

Swiftb0y replied:
eslint broke when I removed it. Seems to me like some obscure bug but idk.

Holzhaus committed:
Beats: Add new unified beats implementation

This is supposed to combine the beatmap/beatgrid classes into a single,
versatile implementation.

Holzhaus committed:
Beats: Use new beats implementation for beatgrids

Holzhaus committed:
Beats: Implement methods to create object from const tempo or beat vector

Holzhaus committed:
Beats: Implement BPM scaling

Holzhaus committed:
Beats: Remove old BeatMap/BeatGrid classes

Holzhaus committed:
BeatMapTest: Fix broken TestNthBeat test

tcoyvwac committed:
Switch to automatic datatype: std::vector

Notes:
* m_pFifoChunkBuffer was actualy an array pointer; was not deleted
safely. Refactor fixes memory leak.
* Used void*[] implicit conversions for protobuf.
* Used "auto _ = std::vector{}" style for some assignments.
* Default'ed empty virtual destructors and fully removed other empty
destructors from codebase.
* Used std::vector<T>.resize() in place of raw-new.

ronso0 committed:
librarycontrol: allow to focus the library widget directly...

without the indirection via sidebar, and without assuming the library is next/after the sidebar.

Holzhaus committed:
Beats: Merge new beats implementation into abstract Beats class

Holzhaus committed:
Beats: Add methods to access underlying data structures as tests

Holzhaus committed:
BeatMapTest: Fix number of beats in scale test

Holzhaus committed:
Beats: Fix Serato BeatGrid import

Holzhaus committed:
BeatsTest: Add more tests for `getBpm` methods

Holzhaus committed:
BeatMapTest: Remove TestBpmAround due to rounding issues

The tests fails although the results are roughly in the same ballpark
(+/- 4 BPM):

    src/test/beatmaptest.cpp:249: Failure
    Expected equality of these values:
      63.937645572318047
      pMap->getBpmAroundPosition( mixxx::audio::kStartFramePos + 4 * approx_beat_length, 4) .value()
        Which is: 64.419153961777965
    src/test/beatmaptest.cpp:254: Failure
    Expected equality of these values:
      118.96668932698844
      pMap->getBpmAroundPosition( mixxx::audio::kStartFramePos + 60 * approx_beat_length, 4) .value()
        Which is: 122.00000000000068
    src/test/beatmaptest.cpp:260: Failure
    Expected equality of these values:
      62.937377309576974
      pMap->getBpmAroundPosition(mixxx::audio::kStartFramePos, 4).value()
        Which is: 60.731050014550149
    src/test/beatmaptest.cpp:262: Failure
    Expected equality of these values:
      118.96668932698844
      pMap->getBpmAroundPosition( mixxx::audio::kStartFramePos + 65 * approx_beat_length, 4) .value()
        Which is: 122.00000000000068

Due to the test design it's not possible to verify these floating point
results manually (on paper), and it's likely that these issues are
caused by rounding the gradually increasing beat lengths.

The BPM calculation methods are already tested in `BeatsTest`, so we
just remove this.

Holzhaus committed:
Beats: Remove legacy BeatIterator class in favor of std::iterator

The custom, legacy iterator API was only used for rendering beats on the
waveform and for importing beats from Serato BeatGrids.

Since the new iterator API is way more powerful and can be used
with the iterator-based functions from `<algorithm>`, it makes sense to
replace the only occurrence and then remove the old API entirely.

ronso0 committed:
library: add CO to read and set the currently focused library widget

Be-ing committed:
GH Actions: lower compiler cache size to 0.5 GB

Hopefully this will result in less frequent cache misses.

Be-ing committed:
Merge pull request #4426 from daschuer/effects_refactoring_loaded_effect

Effects refactoring: loaded effect

Be-ing committed:
EffectChain: name ControlObjects consistently with chain_preset

Be-ing committed:
EffectChain: make setting chain_preset_selector to 0 a no-op

daschuer committed:
Merge pull request #4457 from Be-ing/chain_preset_selector_0

EffectChain: make setting chain_preset_selector to 0 a no-op

ronso0 committed:
Search: fix popup highlight

ronso0 committed:
Search: Del or Backspace in popup deletes query

ronso0 committed:
Search: add ControlPushButton to remove selected query

ronso0 committed:
Search: add isEnabled() tests

ronso0 committed:
Search: save queries across restarts and when switching skins

ninomp committed:
WTrackMenu: Extract hide/remove key sequence to a constant

Holzhaus committed:
Beats: Skip additional marker insertion if beats are beatgrid-compatible

Without this patch, every beats instance created from a beats vector
would be stored a `BeatMap` in the database, even if a `BeatGrid` would
suffice.

This also "fixes" the broken `EngineSyncTest.HalfDoubleConsistency` test
which suffered from a minor rounding issue in
`Beats::getBpmAroundPosition`, which is causes by the initial play
position of an `EngineBuffer` being set to
`std::numeric_limits<double>::lowest()`, which leads to slightly
inaccurate floating point results in `getBpmAroundPosition()`.

Because this commit makes the beats instance used in the test have a
constant tempo, this "fixes" the issue because the
`getBpmAroundPosition()` can exit early and just return the (accurate)
end marker BPM.

Be-ing committed:
Merge pull request #4456 from Be-ing/chain_preset_co

EffectChain: name ControlObjects consistently with chain_preset

ninomp committed:
WTrackTableView: Use separate i18n string for each track removal message

ninomp committed:
WTrackTableView: Use curly braces for single-statement ifs

ninomp committed:
WTrackTableView: Deduplicate local variable declaration

Holzhaus committed:
Merge pull request #4452 from Be-ing/cache_size

GH Actions: lower compiler cache size to 0.5 GB

ronso0 committed:
Search: don't store duplicates in slotSaveSearch()

daschuer committed:
Rename variables of type EnigineParameter to engineParameters

This finishes the refactoring started in https://github.com/mixxxdj/mixxx/commit/33405b2227a3a04f4909981fcda6b09787b821b1

daschuer committed:
Initialize sampleRate

daschuer committed:
Added missing virtual destructor in EffectRequestHandler

Holzhaus committed:
Merge pull request #4459 from daschuer/effects_refactoring_engineParameters

Rename variables of type EnigineParameter to engineParameters

uklotzde committed:
Merge pull request #4341 from tcoyvwac/fix/remove-raw-pointer-types

Removed raw array pointer types for std::vector

ninomp committed:
WTrackMenu: Use auto instead of QKeySequence

ninomp committed:
WTrackTableView: Make hide/remove confirmation messages consistent

Holzhaus committed:
Merge pull request #4369 from ronso0/focus-library-directly

Focus library directly

daschuer committed:
Improve Text for saving a effect chain as a preset

ronso0 committed:
Search: also save search on focus out

ronso0 committed:
Search: don't update empty combobox when removing list items

Commit:
Merge branch '2.3' of github.com:mixxxdj/mixxx into updateaudible

Commit:
Update Audible: Better location for call

Holzhaus committed:
AnalyzerBeats: Fix wrong log message about detected BPM

Holzhaus committed:
Beats: Fix wrong comment stating that getBpmInRage returns average BPM

Holzhaus committed:
Beats: Clarify that getBpmAroundPosition returns the arithmetic average

daschuer committed:
Initalize RampedSample::currentValue

daschuer committed:
Merge pull request #4361 from Holzhaus/getbpminrange

Beats: Refactor getBpm() into getBpmInRange(startPos, endPos)

uklotzde committed:
Merge pull request #4330 from ninomp/del-key-to-remove-tracks

Allow to remove tracks by pressing Delete key

Be-ing committed:
Merge pull request #4462 from daschuer/effects_refactoring_1

Effects refactoring minor fixes

toszlanyi committed:
add supported file types for Engine DJ (#4463)

* add supported file types for Engine DJ

Holzhaus committed:
Beats: Don't return arithmetic average from getBpmInRange()

This method should return the predominant BPM value instead.

Be-ing committed:
Merge pull request #4466 from daschuer/effects_refactoring_4

Improve Text for saving a effect chain as a preset

daschuer committed:
Merge remote-tracking branch 'upstream/main' into effects_refactoring

ronso0 committed:
effects preferences: fix update()

ronso0 committed:
effects prefs: use '---' instead of 'None' to spot empty slots easier

Be-ing committed:
Merge pull request #4469 from ronso0/effects-pref-none

effects prefs: use '---' instead of 'None' to spot empty slots easier

Be-ing committed:
DlgPrefEffects: fix preset removed from list when canceling delete

fixes https://bugs.launchpad.net/mixxx/+bug/1947922

ronso0 committed:
effects preferences: allow only single selection in each tab

Be-ing committed:
fix setting of effect parameters programatically

fixes https://bugs.launchpad.net/mixxx/+bug/1947570

Be-ing committed:
Merge pull request #4468 from ronso0/effects-pref-fixes

Effects preferences fixes

Be-ing committed:
EffectChain: fix aliasing to nonexistent ControlObject

Be-ing committed:
ControlDoublePrivate: change warnings to debug assertions

Swiftb0y committed:
Merge pull request #4472 from Be-ing/wrong_alias

EffectChain: fix aliasing to nonexistent ControlObject

Swiftb0y committed:
Merge pull request #4470 from Be-ing/preset_delete

DlgPrefEffects: fix preset removed from list when canceling delete

Commit:
Merge pull request #4366 from daschuer/fuzzy_beats

Fuzzy beats fix.

daschuer committed:
Merge pull request #4473 from Be-ing/alias_debug_assert

ControlDoublePrivate: change warnings to debug assertions

Commit:
Sync Lock: make sure to reinit leader params after scratching ends

Fixes an issue where scratching the leader while another stopped deck was present would cause the leader to change pitch to resync.

Commit:
Sync Lock: Have bpmcontrol override notifySeek

uklotzde committed:
TrackDAO: Load a single track by URL

Commit:
Disconnect connections from controllers when deleting controller prefs

Leaving the stale connections can cause segfaults on exit.

This fixes https://bugs.launchpad.net/mixxx/+bug/1946581

Commit:
Controller disconnect: be more restrictive about what we disconnect

Commit:
controller disconnect: store an explicit list of connections

Be-ing committed:
Merge pull request #4475 from uklotzde/track-by-url

TrackDAO: Load a single track by URL

uklotzde committed:
TrackAnalysisScheduler: Delete unused function

uklotzde committed:
AnalyzerWaveform: Delete obsolete include directive

uklotzde committed:
TrackModel: Add getTrackUrl()

uklotzde committed:
TrackModel: Delete unused member functions

uklotzde committed:
TrackModel: Replace getTrackLocation() with getTrackUrl()

Be-ing committed:
Merge pull request #4477 from uklotzde/trackmodel-url

TrackModel::getTrackUrl()

uklotzde committed:
TrackAnalysisScheduler: Remove dependency on Library

Be-ing committed:
Merge pull request #4478 from uklotzde/trackanalysisscheduler

[Qt6] TrackAnalysisScheduler: Remove dependency on Library

uklotzde committed:
Qt6: Re-enable Waveform and WaveformFactory classes

uklotzde committed:
TrackAnalysisScheduler: Remove dependency on TrackDAO

Commit:
Sync Lock: Make sure to update the leader beat distance after a seek

Because of half/double math, a phased seek might cause the Leader to be 0.5 beats off from where it used to be. Therefore after a seek, make sure to update the leader beat distance with the new value.

Be-ing committed:
Merge pull request #4479 from uklotzde/qt6-waveform

Qt6: Re-enable Waveform and WaveformFactory classes

Be-ing committed:
Merge pull request #4480 from uklotzde/trackanalysisscheduler

TrackAnalysisScheduler: Remove dependency on TrackDAO

uklotzde committed:
Finalize implementing class and methods

uklotzde committed:
Add TODO for possible replacement of a plain pointer

daschuer committed:
Merge pull request #4471 from Be-ing/main_eq

fix setting of effect parameters programatically

daschuer committed:
Initalize WEffectCahinPresetButton::m_iChainNumber

daschuer committed:
Improve description why we don't want to rebased PRs (#4464)

Improve description why we don't want to rebase PRs

daschuer committed:
Improve code style in dlgprefeq.cpp

daschuer committed:
rename kEffectForGroupPrefix

daschuer committed:
Make use of QStringLiteral in dlgprefeq.cpp

daschuer committed:
Added missing null check in loadChainPreset()

daschuer committed:
Clear effect chain when trying to load one that no longer exists

daschuer committed:
rename kConfigGroup

Be-ing committed:
Merge pull request #4483 from daschuer/effects_refactoring_lp1948606

Effects refactoring: fix segfault

uklotzde committed:
Merge pull request #4482 from daschuer/effects_refactoring_6

Effects refactoring minor changes

Swiftb0y committed:
Merge pull request #4481 from uklotzde/trackanalysisscheduler

TrackAnalysisSchedulerEnvironmentImpl: Finalize implementing class and methods

Holzhaus committed:
Merge branch 'main' of github.com:mixxxdj/mixxx into newbeats

ywwg committed:
controller disconnect: use the list of controllers to determine what to disconnect

This is safe code for now, but may not be in the future. Added some comments as hints to future developers trying to add hotplug so they don't trip over this code.  (There is no way to write this code currently that would be future-proof)

daschuer committed:
Rename presetNameChanged() to chainPresetChanged(), because not only the name changes

daschuer committed:
Added new signal EffectChainPresetManager::effectChainPresetRenamed and make use of it.

daschuer committed:
Merge pull request #4005 from ywwg/synclock-06-scratch-phase

Sync Lock: make sure to reinit leader params after scratching ends

daschuer committed:
Merge pull request #4255 from Holzhaus/newbeats

Add new beats implementation

Be-ing committed:
Merge pull request #4484 from daschuer/effects_refactoring_lp1948567

Fix empty effect chain selector after renaming chain presets

Be-ing committed:
Merge pull request #4467 from mixxxdj/effects_refactoring

Effects refactoring

Holzhaus committed:
Beats: Remove leftover qWarning() from debugging

Swiftb0y committed:
Merge pull request #4488 from Holzhaus/newbeats-leftover-warning

Beats: Remove leftover qWarning() from debugging

ywwg committed:
controller disconnect : Make sure to associate the connection with the parent object

Holzhaus committed:
Beats: Add comments for iterator methods (cbegin/cend/cearliest/clatest)

ronso0 committed:
Merge remote-tracking branch 'mixxx/main' into lib-selection-state

ronso0 committed:
DlgAutoDJ: use setFocus() instead of keypress emulation

daschuer committed:
Reintroduce the Quick effect selectors in the EQ preferences

We need that to merge the effects_refactoring branch without a regression until the Skin effect GUI is in place. https://bugs.launchpad.net/mixxx/+bug/1948537

daschuer committed:
Merge pull request #4490 from ronso0/autodj-setfocus

DlgAutoDJ: use setFocus() instead of keypress emulation

ronso0 committed:
Merge branch 'main' into lib-selection-state

daschuer committed:
Merge pull request #4458 from ronso0/search-delete-saved-queries

Search: save queries across restarts, related tweaks & fixes

daschuer committed:
Merge pull request #4363 from ywwg/updateaudible

Sync Lock: Perform updateAudible work in process call

daschuer committed:
Merge pull request #4411 from Holzhaus/newbeats-iterator

Beats: Remove legacy BeatIterator class in favor of std iterator

ronso0 committed:
Merge pull request #4486 from daschuer/effects_refactoring_lp1948537

Reintroduce the Quick effect selectors in the EQ preferences

Swiftb0y committed:
midi-components-0.0.js: Pot support for arbitrary maximums in 7-/14-bit handlers

previously, it was not possible for 14-bit controls to have maxima
that are not (2^14)-1. Some controller might only have 3-bits of
resolution in their MSB. This commit allows for arbitrary precision
in MSB and LSB by specifying an appropriate value as the components
max property.

Swiftb0y committed:
.pre-commit-config.yaml: disallow commiting to version branches

before this commit, it was possible to accidentally commit to
the stable version branches (such as 2.3). If this was done
by a member with push access, they could push untested commits
into production on accident. If this was done by a git newbie,
they would mess up their local branch and deal with conflicts
when pulling their branch later. If commiting to the branch
is justified, the check can be bypassed by
`SKIP=no-commit-to-branch git commit`

Swiftb0y committed:
Revert "midi-components-0.0.js: Pot support for arbitrary maximums in 7-/14-bit handlers"

This reverts commit 4ec46c62eec26e4442e7a8b8107da7a9fffddebb.

daschuer committed:
Merge pull request #4494 from Swiftb0y/pre-commit-disallow-committing-to-version-branches

.pre-commit-config.yaml: disallow commiting to version branches

ronso0 committed:
Effect chain selector: allow individual text alignment

ronso0 committed:
Effect chain selector: individual elide for button & popup text

ronso0 committed:
Effect selector: elide item texts depending on popup width

In oficial skins the popup is much wider than the selector which
previosly resulted in unnecessarily shortened effect names in the popup.

qgazq committed:
Add environment var CMAKE_PREFIX_PATH back in so a valid json file is produced

qgazq committed:
chan tab to a space to match

qgazq committed:
Add quotes to match

ywwg committed:
Merge branch '2.3'

Holzhaus committed:
Beats: Rename cbegin/cend to cfirstmarker/clastmarker

Holzhaus committed:
Beats: Let clastmarker() point to the marker, not the beat behind it

Holzhaus committed:
Beats: Rename cearliest/clatest to cbegin/cend

ronso0 committed:
Browse model: inline clearPath()

ronso0 committed:
library models: move modelKey prefixes to namespaces

daschuer committed:
Merge pull request #4498 from Holzhaus/beats-cbegin-cend

Rename Beats iterator methods

Holzhaus committed:
Beats: Round positions to lower frame boundary on translate

This fixes the following debug assertion:

    DEBUG ASSERT: "!m_endMarkerPosition.isFractional()" in function mixxx::Beats::Beats(std::vector<mixxx::BeatMarker>, mixxx::audio::FramePos, mixxx::Bpm, mixxx::audio::SampleRate, const QString&) at /home/jan/Projects/mixxx/src/track/beats.h:177

ronso0 committed:
libraryvie.h: satisfy clang-format

ronso0 committed:
LateNight: add Quick Effect selector

ronso0 committed:
skins: add default effect menu styles

Holzhaus committed:
Merge pull request #4497 from qgazq/main

Fixed invalid CMakeSettings.json generation

Holzhaus committed:
Beats: Remove leftover qWarning from debugging

Holzhaus committed:
Merge pull request #4476 from ywwg/disconnect-controllers

Disconnect controller qt connections

Be-ing committed:
Merge pull request #4500 from Holzhaus/newbeats-qwarning

Beats: Remove leftover qWarning from debugging

ronso0 committed:
Effect chain selector: pass focus to library after selecting effect

ronso0 committed:
Pref > Library: add tabstop for Serato Metadata

ronso0 committed:
Pref > Library: fix reset/default search debounce timeout + update

daschuer committed:
Merge pull request #4501 from ronso0/fxrfct-chain-selector

Tweak WEffectChainPresetSelector

daschuer committed:
Merge pull request #4502 from ronso0/fxrfct-latenight

LateNight: add Quick Effect selector

ronso0 committed:
Merge pull request #4504 from ronso0/pref-lib-tab-order

Preferences > Library: fix search timeout reset + tab order

Holzhaus committed:
LegacySkin: Add missing include for QRegularExpression

Should fix a build error:

    /var/tmp/portage/media-sound/mixxx-9999/work/mixxx-9999/src/skin/legacy/legacyskin.cpp:8:40: error: variable ‘const QRegularExpression {anonymous}::kMinSizeRegExp’ has initializer but incomplete type
        8 | const QRegularExpression kMinSizeRegExp(QStringLiteral("<MinimumSize>(\\d+), *(\\d+)<"));
          |                                        ^
    /var/tmp/portage/media-sound/mixxx-9999/work/mixxx-9999/src/skin/legacy/legacyskin.cpp:9:37: error: variable ‘const QRegularExpression {anonymous}::kDigitRegex’ has initializer but incomplete type
        9 | const QRegularExpression kDigitRegex(QStringLiteral("\\d"));
          |                                     ^
    /var/tmp/portage/media-sound/mixxx-9999/work/mixxx-9999/src/skin/legacy/legacyskin.cpp: In member function ‘virtual bool mixxx::skin::legacy::LegacySkin::fitsScreenSize(const QScreen&) const’:
    /var/tmp/portage/media-sound/mixxx-9999/work/mixxx-9999/src/skin/legacy/legacyskin.cpp:93:33: error: aggregate ‘QRegularExpressionMatch match’ has incomplete type and cannot be defined
       93 |         QRegularExpressionMatch match;
          |                                 ^~~~~

Reported here: https://github.com/mixxxdj/mixxx/pull/4289#issuecomment-962651800

Be-ing committed:
Merge pull request #4506 from Holzhaus/legacyskin-qregularexpression-incomplete-type

LegacySkin: Add missing include for QRegularExpression

daschuer committed:
Merge pull request #4509 from ronso0/knob-reset-double-click

double click to reset knobs and sliders

Holzhaus committed:
Beats: Improve position variable name in `tryTranslate`

Holzhaus committed:
Beats: Use "last marker" term instead of "end marker"

This was discussed on Zulip: https://mixxx.zulipchat.com/#narrow/stream/109171-development/topic/beat.20grid.20revamp.20re.3A.20pr.20.232961/near/260871852

daschuer committed:
Merge pull request #4499 from Holzhaus/newbeats-translate-assert

Beats: Round positions to lower frame boundary on translate

uklotzde committed:
Adjust the download directory and name of external dependencies

The file name in the URL doesn't match the actual file name that
is downloaded from GitHub.

uklotzde committed:
Adjust the download directory and name of external dependencies

The file name in the URL doesn't match the actual file name that
is downloaded from GitHub.

Swiftb0y committed:
.eslintrc.json: adjust eslint rules to enforce readable chain syntax

previously, the eslint hook would enforce that the dot of multi-
line chaining was attached to the object instead of the property
thus formatting the first example into the second which I
find less readable.

```javascript
const a = _({})
    .mapKeys(...)
    .filter(...)
    .value();
```
```javascript
const a = _({}).
    mapKeys(...).
    filter(...).
    value();
```

Holzhaus committed:
Merge pull request #4512 from Swiftb0y/eslint-chained-method-calls

.eslintrc.json: adjust eslint rules to enforce readable chain syntax

Holzhaus committed:
Beats: Add some comments

ronso0 committed:
Search: Ctrl+F in focused search box selects the entire search string

ywwg committed:
Merge pull request #4510 from Holzhaus/newbeats-lastmarker

Beats: Use "last marker" term instead of "end marker"

uklotzde committed:
Update libkeyfinder versions

Be-ing committed:
Merge pull request #4514 from uklotzde/cmake-external-downloads-libdjinterop

CMake: Adjust the download directory and name of external dependencies Pt. 2

Be-ing committed:
Merge pull request #4511 from uklotzde/cmake-external-downloads

CMake: Adjust the download directory and name of external dependencies

Holzhaus committed:
Roland DJ-505: Use let/const instead of var

ronso0 committed:
LateNight: fixup width of Quick Effect popup

ywwg committed:
Merge pull request #4515 from ronso0/search-ctrl-f

Search: Ctrl+F in focused search box selects the entire search string

Be-ing committed:
Merge remote-tracking branch 'upstream/2.3'

Be-ing committed:
Merge pull request #4518 from ronso0/late-quickeffect-popup-width

LateNight: fixup width of Quick Effect popup

Holzhaus committed:
Update res/controllers/Roland_DJ-505-scripts.js

Co-authored-by: Swiftb0y <12380386+Swiftb0y@users.noreply.github.com>

Be-ing committed:
Merge pull request #4328 from fellock/2.3

Changed beatloop_activate button behaviour to match lp1876003

Swiftb0y committed:
Merge pull request #4517 from Holzhaus/roland-dj-505-var-const

Roland DJ-505: Use let/const instead of var

Be-ing committed:
HidController: return QByteArrays for get[Input/Feature]Report

for compatibility with HidController::poll

Swiftb0y commented:
Whats left to do here?

Be-ing commented:
will take another look soon. I want to get this merged for 2.4 beta

Swiftb0y commented:
I still wouldn't really want to ship it for 2.4 but getting it into main ASAP would be good regardless.

ronso0 committed:
Merge remote-tracking branch 'mixxx/main' into remove-from-disk

ronso0 committed:
create QDialogs on the stack

ronso0 committed:
refine comment in RemoveTrackFilesFromDiskTrackPointerOperation

ronso0 committed:
instantiate dialog ingredients just before they're needed

Holzhaus committed:
ControllerDebug: Remove singleton in favor of categorized logging

daschuer committed:
Merge pull request #4525 from ronso0/skin-control-info

Use info not warning for skin COs

Holzhaus committed:
common-controller-scripts.js: Make engine.log use console.log internally

Holzhaus committed:
Logging: Filter all debug messages from the controller category

daschuer committed:
This reverts a comment with a wrong assumption

introduced in #2635 and commit 8ee1dd3bd74de976f460742cca3dbd3543658024.

Holzhaus committed:
Merge pull request #4527 from daschuer/commentRevert

Comment revert

Swiftb0y committed:
Merge pull request #4524 from Holzhaus/controller-console-log

Controllers: Use console.log instead of engine.log

Holzhaus committed:
Logging: Improve documentation for special --controller-debug logging

Holzhaus committed:
PortMidiController: Check for outputDeviceInfo nullptr in constructor

daschuer committed:
Merge pull request #4523 from Holzhaus/logging-controller-debug

Controller logging categories (with --controllerDebug support)

Be-ing committed:
HidController: take QByteArray for sendFeatureReport

Be-ing committed:
HidController: avoid extra copy of data

ronso0 committed:
WEffectChainPresetSelector: don't set elide padding in c++

daschuer committed:
Merge pull request #4177 from ronso0/lib-selection-state

Fix library track selection, save model state

Be-ing committed:
Merge pull request #4532 from ronso0/fxpresetselector-elide

WEffectChainPresetSelector: don't set elide padding in c++

uklotzde committed:
Add missing const qualifiers to isDirty()

uklotzde committed:
TrackDAO: Fix semantics of marking a track clean after update

daschuer committed:
Merge pull request #3550 from Be-ing/remove_scons_cpack

CMake: remove obsolete references to files generated by scons

rryan committed:
Merge pull request #1966 from Be-ing/audio_engine_cleanup

remove autogenerated code from ChannelMixer

uklotzde committed:
ChannelMixer: Use range-based for loops

Be-ing committed:
Merge pull request #4535 from uklotzde/audio_engine_cleanup

ChannelMixer: Use range-based for loops

Holzhaus committed:
Merge pull request #4534 from uklotzde/is-dirty-const

Fix isDirty() signature and semantics

uklotzde committed:
Merge pull request #4533 from Be-ing/rename_maybe

rename maybeStartOrStopPolling to pollIfAnyControllersOpen

Be-ing committed:
gitignore: change /build to /build*

This allows for multiple CMake build directories, for example
build-qt5, build-qt6, build-vcpkg.

daschuer committed:
Merge pull request #4537 from Be-ing/gitignore-build

gitignore: change /build to /build*

ronso0 committed:
remove unused includes QDesktop/QStandardPaths

ronso0 committed:
remove files from disk: de-duplicate file list

ronso0 committed:
remove files from disk: fix comment typo

ronso0 committed:
remove files from disk: replace length() > 0 with isEmpty()

ronso0 committed:
WTrackTableView: use QModelIndexList::isEmpty(), return early

ronso0 committed:
const ref purge/keep track lists, more const

uklotzde committed:
Merge pull request #3212 from ronso0/remove-from-disk

WTrackMenu: add Remove From Disk action

uklotzde committed:
Update Benchmark library to v1.6.0

uklotzde committed:
WTrackMenu: Add missing feature to all features enum variant

Holzhaus committed:
Merge pull request #4541 from uklotzde/UpdateReplayGain

WTrackMenu: Add missing feature to all features enum variant

Holzhaus committed:
Merge pull request #4540 from uklotzde/benchmark-v1.6.0

Update vendored Benchmark library to v1.6.0

uklotzde committed:
Merge branch '2.3' of git@github.com:mixxxdj/mixxx.git

uklotzde committed:
WTrackMenu::featureIsEnabled(): Fix unhandled case

daschuer committed:
Merge pull request #4543 from uklotzde/WTrackMenu_featureIsEnabled

WTrackMenu::featureIsEnabled(): Fix unhandled case

ronso0 committed:
keep inapplicable sound config until new config is accepted

daschuer committed:
Merge pull request #4544 from ronso0/sound-config-error

don't wipe inapplicable sound config immediately

ronso0 committed:
Merge remote-tracking branch 'mixxx/2.3' into main

fwcd committed:
Add Big Sur-style rounded rectangle icon for macOS

- Add initial macOS Big Sur icon draft
- Replace macOS icon lens flare with brighter gradient
- Add drop shadow to macOS icon
- Update macOS ICNS files
- Remove hidden layers from new macOS icon

fwcd committed:
Define CFBundleName in macOS bundle

Holzhaus committed:
Merge pull request #4546 from fwcd/cfbundlename

Define `CFBundleName` in macOS bundle

Swiftb0y committed:
Merge pull request #4521 from Be-ing/hid_getInputReport_bytearray

HidController: return QByteArrays for get[Input/Feature]Report

ronso0 committed:
Merge pull request #4545 from fwcd/macos-bigsur-icon

Add a Big Sur-style rounded rectangle icon on macOS

ywwg committed:
Minimixxx controller config (#4350)

* MiniMixxx: Initial implementation

* MiniMixxx: Start implementing buttons

* MiniMixxx: Implement layers

* MiniMixxx: relative pitch slider mode

* MiniMixxx: wheel spinnies!

* MiniMixxx: forgot to hook up a connection

* MiniMixxx: small light tweak

* MiniMixxx: sampler layer

* MiniMixxx: Fix some layer switching issues

* MiniMixxx: library mode

* MiniMixxx: add fx layer

oh no I have no room for hotcues!

* MiniMixxx: rearrange file

* MiniMixxx: factor out vu meter color

* MiniMixxx: Add hotcue layer

Because we are out of buttons, hotcues are activated for both decks at once.  That seems ok.

MiniMixxx is now feature complete!

* MiniMixxx: allow for scrubbing through track in library mode

* MiniMixxx: Add newline to satisfy linter

* MiniMixxx: Factor out colors

* MiniMixxx: Add main output adjustments

* MiniMixxx: trying to get consistent behavior with layers

* MiniMixxx: cleanup

* MiniMixxx: more cleanup

* MiniMixxx: Apply eslint

* MiniMixxx: cleanup and fix comments

* MiniMixxx: Upgrade to javascript classes

* MiniMixx: fix sync button push-and-lock

* MiniMixxx: Use newer engine connection API

* MiniMixxx: use proper debug output

* Minimixxx: make it possible to expand tree items in library browsing

* MiniMixxx: bind booth volume adjustment

Also create mode where pitch sliders don't do anything *unless* shift is held.
This prevents mixtakes

* minimixxx: adjust jog sensitivity

* MiniMixxx: make righthand library encoder scroll horizontally

also some minor tweaks

* Minimixxx: fix typo

* raspi: make hotcues the non-shift action

* Minimixxx: rename files to be consistent

* Minimixxx: use switches where possible

* Update res/controllers/Yaeltex MiniMixxx.midi.xml

Co-authored-by: Jan Holthuis <holthuis.jan@googlemail.com>

* Update res/controllers/Yaeltex-MiniMixxx-scripts.js

Co-authored-by: Be <be.0@gmx.com>

* Minimixxx: revert bad change (var was undefined)

* Minimixxx: disclaimer

Co-authored-by: Jan Holthuis <holthuis.jan@googlemail.com>
Co-authored-by: Be <be.0@gmx.com>

Holzhaus committed:
Merge pull request #4542 from ronso0/include-cleanup-desktop.stadartpaths

remove unused includes QDesktop/QStandardPaths
